packages/gatsby/index.d.ts

```diff
@@ -11,24 +14,859 @@ export {
+export interface PreprocessSourceArgs extends ParentSpanPluginArgs {
+  filename: string;
+  contents: string;
+}
+
+export interface ResolvableExtensionsArgs extends ParentSpanPluginArgs {
+  traceId: string;
+}
+
+export interface SetFieldsOnGraphQLNodeTypeArgs extends ParentSpanPluginArgs {
+  type: {
+    name: string;
+    nodes: any[];
+  };
+  traceId: string;
+}
+
+export interface SourceNodesArgs extends ParentSpanPluginArgs {
+  traceId: string;
+  waitForCascadingActions: boolean;
+}
+
+export interface PreRenderHTMLArgs extends NodePluginArgs {
+  getHeadComponents: any[];
+  replaceHeadComponents: Function;
+  getPreBodyComponents: any[];
+  replacePreBodyComponents: Function;
+  getPostBodyComponents: any[];
+  replacePostBodyComponents: Function;
+}
+
+export interface RenderBodyArgs extends NodePluginArgs {
+  pathname: string;
+  setHeadComponents: Function;
+  setHtmlAttributes: Function;
+  setBodyAttributes: Function;
+  setPreBodyComponents: Function;
+  setPostBodyComponents: Function;
+  setBodyProps: Function;
+}
+
+export interface ReplaceRendererArgs extends NodePluginArgs {
+  replaceBodyHTMLString: Function;
+  setHeadComponents: Function;
+  setHtmlAttributes: Function;
+  setBodyAttributes: Function;
+  setPreBodyComponents: Function;
+  setPostBodyComponents: Function;
+  setBodyProps: Function;
+}
+
+export interface WrapPageElementNodeArgs extends NodePluginArgs {
+  element: object;
+  props: object;
+  pathname: string;
+}
+
+export interface WrapRootElementNodeArgs extends NodePluginArgs {
+  element: object;
+}
+
+export interface ParentSpanPluginArgs extends NodePluginArgs {
+  parentSpan: object;
+}
+
+export interface NodePluginArgs {
+  pathPrefix: string;
+  boundActionCreators: Actions;
+  actions: Actions;
+  loadNodeContent: Function;
+  store: Store;
+  emitter: EventEmitter;
+  getNodes: Function;
+  getNode: Function;
+  getNodesByType: Function;
+  hasNodeChanged: Function;
+  reporter: Reporter;
+  getNodeAndSavePathDependency: Function;
+  cache: Cache;
+  createNodeId: Function;
+  createContentDigest: Function;
+  tracing: Tracing;
+  [key: string]: unknown;
+}
+
+export interface Actions {
+  /** @see https://www.gatsbyjs.org/docs/actions/#deletePage */
+  deletePage: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#createPage */
+  createPage(args: CreatePageArgs, option?: ActionOptions): void;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#deletePage */
+  deleteNode(args: DeleteNodeArgs, option?: ActionOptions): void;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#deleteNodes */
+  deleteNodes: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#createNode */
+  createNode(node: Node, options?: ActionOptions): void;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#touchNode */
+  touchNode: Function;
+  
+  /** @see https://www.gatsbyjs.org/docs/actions/#createNodeField */
+  createNodeField(args: CreateNodeFieldArgs, options?: ActionOptions): void;
```

Comment:
I could not find `CreateNodeFieldArgs`
```suggestion
  createNodeField(args: {node: Node, fieldName?: string, fieldValue?: string, name?: string, value: any}, plugin?: ActionPlugin, options?: ActionOptions): void;
```